rotkehlchen/tests/unit/decoders/test_llamazip.py

```diff
@@ -0,0 +1,418 @@
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=optimism_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp, fee_amount, spend_amount, receive_amount = TimestampMS(1724074841000), '0.000000292773423793', '24.786522600837181987', '24.780601'  # noqa: E501
+    assert events == [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(fee_amount)),
+            location_label=optimism_accounts[0],
+            notes=f'Burned {fee_amount} ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.SPEND,
+            asset=Asset('eip155:10/erc20:0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1'),
+            balance=Balance(amount=FVal(spend_amount)),
+            location_label=optimism_accounts[0],
+            notes=f'Swap {spend_amount} DAI in LlamaZip',
+            counterparty=CPT_LLAMAZIP,
+            address=string_to_evm_address('0xbf16ef186e715668AA29ceF57e2fD7f9D48AdFE6'),
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=2,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.RECEIVE,
+            asset=Asset('eip155:10/erc20:0x7F5c764cBc14f9669B88837ca1490cCa17c31607'),
+            balance=Balance(amount=FVal(receive_amount)),
+            location_label=optimism_accounts[0],
+            notes=f'Receive {receive_amount} USDC.e as the result of a swap in LlamaZip',
+            counterparty=CPT_LLAMAZIP,
+            address=OPTIMISM_ROUTER_ADDRESSES[0],
+        ),
+    ]
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('arbitrum_one_accounts', [['0x635Cb8149C292Ff96F71a1a49120D04053c7eE7A']])
+def test_llamazip_arbitrum_swap_token_to_eth(database, arbitrum_one_inquirer, arbitrum_one_accounts):  # noqa: E501
+    tx_hash = deserialize_evm_tx_hash('0x5a3e6748dd62b943918508a7495df0dc481a054af9dc607a7b85f167a9cc54c2')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=arbitrum_one_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp, fee_amount, spend_amount, receive_amount = TimestampMS(1679552792000), '0.0000306163', '36', '0.020467522941555658'  # noqa: E501
+    a_usdce = Asset('eip155:42161/erc20:0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8')
```

Comment:
nitpick: combine those in one line as definitions of the test

```suggestion
    timestamp, fee_amount, spend_amount, receive_amount = TimestampMS(1679552792000), '0.0000306163', '36', '0.020467522941555658'  # noqa: E501
    a_usdce = Asset('eip155:42161/erc20:0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8')
```